src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -931,3 +907,147 @@ def has_bluestore_label(device_path):
         logger.info(f'{device_path} is a directory, skipping.')
 
     return isBluestore
+
+def _dd(device: str, count: int, skip: int = 0) -> str:
+    """Read bytes from a device
+
+    Args:
+        device (str): The device to read bytes from.
+        count (int): The number of bytes to read.
+        skip (int, optional): The number of bytes to skip at the beginning. Defaults to 0.
+
+    Returns:
+        str: A string containing the read bytes.
+    """
+    result: str = ''
+    try:
+        with open(device, 'rb') as b:
+            b.seek(skip)  # skip the first 4096 bytes
+            data: bytes = b.read(count)
+            result = data.decode('utf-8').replace('\x00', '')
+    except OSError:
+        logger.warning(f"Can't read from {device}")
+        pass
+    except UnicodeDecodeError:
+        pass
+    except Exception:
+        raise
+
+    return result
+
+def get_bluestore_header(device: str) -> Dict[str, Any]:
+    """Retrieve BlueStore header information from a given device.
+
+    This function retrieves BlueStore header information from the specified 'device'.
+    It first checks if the device exists. If the device does not exist, a RuntimeError
+    is raised. Then, it calls the 'ceph-bluestore-tool' command to show the label
+    information of the device. If the command execution is successful, it parses the
+    JSON output containing the BlueStore header information and returns it as a dictionary.
+
+    Args:
+        device (str): The path to the device.
+
+    Raises:
+        RuntimeError: If the specified device does not exist.
+
+    Returns:
+        Dict[str, Any]: A dictionary containing BlueStore header information.
+    """
+    data: Dict[str, Any] = {}
+
+    if not os.path.exists(device):
+        raise RuntimeError(f"{device} doesn't exist")
+
+    out, err, rc = process.call([
+        'ceph-bluestore-tool', 'show-label',
+        '--dev', device], verbose_on_failure=False)
+    if rc:
+        logger.debug('device {} is not BlueStore; ceph-bluestore-tool failed to get info from device: {}\n{}'.format(device, out, err))
```

Comment:
NIT: you were converting other, similar messages to f-strings elsewhere in the PR, so we might just want to make this an f-string as well

Reply:
true